packages/big-design/src/components/Checkbox/Checkbox.tsx

```diff
@@ -61,15 +64,18 @@ const RawCheckbox: React.FC<CheckboxProps & PrivateProps> = ({
 
     if (typeof label === 'string') {
       return (
-        <CheckboxLabel
-          aria-hidden={disabled}
-          disabled={disabled}
-          hidden={hiddenLabel}
-          htmlFor={id}
-          id={labelId}
-        >
-          {label}
-        </CheckboxLabel>
+        <>
+          <CheckboxLabel
+            aria-hidden={disabled}
+            disabled={disabled}
+            hidden={hiddenLabel}
+            htmlFor={id}
+            id={labelId}
+          >
+            {label}
+          </CheckboxLabel>
+          {badge ? <Badge marginLeft="xSmall" {...badge} /> : null}
+        </>
```

Comment:
Do you think we'd want the badge to be read along with the label for accessibility 🤔 Having it outside the `label` will remove the badge text from being read along with the label and would require additional keyboard navigation to read it.
```suggestion
          <CheckboxLabel
            aria-hidden={disabled}
            disabled={disabled}
            hidden={hiddenLabel}
            htmlFor={id}
            id={labelId}
          >
            {label}
            {badge ? <Badge marginLeft="xSmall" {...badge} /> : null}
          </CheckboxLabel>
```

Reply:
@chanceaclark thanks for this point. I updated and added a small test